app/src/main/java/com/example/concordia_campus_guide/adapters/RoutesAdapter.java

```diff
@@ -173,24 +186,25 @@ private void setUIShuttle(Route route) {
     private void setUIWalkInTransit(Walk walk) {
         ImageView walkIcon = new ImageView(context);
         walkIcon.setImageResource(R.drawable.ic_directions_walking);
-
+        details.addView(walkIcon);
         TextView durationWalk = new TextView(context);
         durationWalk.setText(walk.getDuration());
         durationWalk.setGravity(Gravity.CENTER);
 
-        details.addView(walkIcon);
         details.addView(durationWalk);
     }
 
     private void setUIBus(Bus bus) {
-        ImageView busIcon = new ImageView(context);
-        busIcon.setImageResource(R.drawable.ic_directions_bus_red);
+        if (bus.getBusNumber() != null) {
```

Comment:
why this null check in particular?

Reply:
Good point I realized that if we're checking transit we don't need to check the bus